### PR TITLE
chore: Replace usage of shell commands with node apis

### PR DESCRIPTION
**Pull request by michaeldowseza**
### Description

- Replace usage of exec cp with fs
- Replace usage of exec rsync wish fs-extra
- Add cross-env to handle env variables in npm scripts
- Add npm-run-all to replace && in npm scripts

Related links, issue #, if available: n/a

### Why
Using shell commands makes our packages OS specific. 

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build-tools/tasks/themeable-source.js [hunk 1]
```diff
@@ -2,14 +2,15 @@
 // SPDX-License-Identifier: Apache-2.0
 const { parallel } = require('gulp');
 const path = require('path');
-const { promises: fsp } = require('fs');
+const { promises: fsp, cpSync } = require('fs');
+const fse = require('fs-extra');
```
Comment (just-boris) [inline build-tools/tasks/themeable-source.js:6]: I do not see this dependency in our `package.json`

**Review by just-boris — COMMENTED** [review]
ok for replacing plain shell commands with node API, but not sure we can market it as "OS support", because if we do not have a CI on that platform it does not count

**Comment by michaeldowseza** [comment]
> ok for replacing plain shell commands with node API, but not sure we can market it as "OS support", because if we do not have a CI on that platform it does not count

I will rename. I guess it only indirectly improves the support.

File: build-tools/tasks/themeable-source.js (hunk 1 above)
Comment (michaeldowseza) [reply build-tools/tasks/themeable-source.js:6]: Hmm, the dependency was available to me. It must be coming in as a transitive. Will add to package json to make it explicit.

**Review by just-boris — APPROVED** [review]

**Pull request merged**